extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/results/ExportResultsFrameProcessor.java

```diff
@@ -167,9 +175,22 @@ private void exportFrame(final Frame frame)
               for (int j = 0; j < exportRowSignature.size(); j++) {
                 String columnName = exportRowSignature.getColumnName(j);
                 BaseObjectColumnValueSelector<?> selector = selectors.get(outputColumnNameToFrameColumnNumberMap.getInt(columnName));
-                exportWriter.writeRowField(columnName, selector.getObject());
+                if (resultsContext == null) {
+                  exportWriter.writeRowField(columnName, selector.getObject());
```

Comment:
Maybe it should fail if resultsContext == null to avoid correctness issues, rather than being backward compatible?

Reply:
I think that makes sense